src/instance-lookup.ts

```diff
@@ -14,7 +14,7 @@ const MYSTERY_HEADER_LENGTH = 3;
 type LookupFunction = (hostname: string, options: dns.LookupAllOptions, callback: (err: NodeJS.ErrnoException | null, addresses: dns.LookupAddress[]) => void) => void;
 
 // Most of the functionality has been determined from from jTDS's MSSqlServerInfo class.
-export async function instanceLookup(options: { server: string, instanceName: string, timeout?: number, retries?: number, port?: number, lookup?: LookupFunction, signal: AbortSignal }) {
+export async function instanceLookup(options: { server: undefined | string, instanceName: string, timeout?: number, retries?: number, port?: number, lookup?: LookupFunction, signal: AbortSignal }) {
```

Comment:
@ruyadorno I think this change is not correct. Logically, the `instanceLookup` can not work without a hostname, so changing the type to allow `undefined` is not correct. Also, casting `options.server` to a string using `String` further down is also not right - if someone would call this method with `undefined`, it would try to send udp messages to the hostname `"undefined"`. 😅 

I think it's better to change the location where this method is called instead (will leave a separate comment for that).